app/services/orchestration_template_dialog_service.rb

```diff
@@ -21,11 +21,25 @@ def add_stack_group(tab, position)
     ).tap do |dialog_group|
       add_tenant_name_field(dialog_group, 0)
       add_stack_name_field(dialog_group, 1)
-      add_on_failure_field(dialog_group, 2)
-      add_timeout_field(dialog_group, 3)
+
+      if template.kind_of?(OrchestrationTemplateAzure)
```

Comment:
This feels to go against [Tell, Don't Ask](https://pragprog.com/articles/tell-dont-ask) principle of OOP.